controllers/managedocs_controller.go

```diff
@@ -436,9 +434,25 @@ func (r *ManagedOCSReconciler) reconcilePhases() (reconcile.Result, error) {
 			r.reconcileStrategy = v1.ReconcileStrategyNone
 		}
 
-		if err := r.get(r.addonParamSecret); err != nil {
+		// Read the secret and store it in addonParams map
+		addonParamSecret := &corev1.Secret{}
+		addonParamSecret.Name = r.AddonParamSecretName
+		addonParamSecret.Namespace = r.namespace
+		if err := r.get(addonParamSecret); err != nil {
 			return ctrl.Result{}, fmt.Errorf("Failed to get the addon param secret, Secret Name: %v", r.AddonParamSecretName)
 		}
+		keys := []string{
+			storageClassSizeKey,
+			enableMCGKey,
+			notificationEmailKeyPrefix + fmt.Sprintf("-%v", 0),
+			notificationEmailKeyPrefix + fmt.Sprintf("-%v", 1),
+			notificationEmailKeyPrefix + fmt.Sprintf("-%v", 2)}
+
+		for _, key := range keys {
+			if value, exists := addonParamSecret.Data[key]; exists {
+				r.addonParams[key] = string(value)
+			}
+		}
```

Comment:
You don't need a key list, just iterate over the map

```suggestion
		for key, value := range addonParamSecret.Data {
			r.addonParams[key] = string(value)
		}
```